src/mssqlProtocolHandler.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as Utils from './models/utils';
+import { IConnectionInfo } from 'vscode-mssql';
+
+enum Command {
+	connect = '/connect',
+	openConnectionDialog = '/openConnectionDialog'
+}
+
+export class MssqlProtocolHandler {
+	constructor() {
+
+	}
+
+	public handleUri(uri: vscode.Uri): IConnectionInfo | undefined {
+		Utils.logDebug(`[MssqlProtocolHandler][handleUri] URI: ${uri.toString()}`);
+
+		switch (uri.path) {
+			case Command.connect:
+				Utils.logDebug(`[MssqlProtocolHandler][handleUri] connect: ${uri.path}`);
+				return this.connect(uri);
+
+			case Command.openConnectionDialog:
+				return undefined;
+
+			default:
+				Utils.logDebug(`[MssqlProtocolHandler][handleUri] Unknown URI path, defaulting to connect: ${uri.path}`);
+				return this.connect(uri);
+		}
+	}
+
+	private connect(uri: vscode.Uri): IConnectionInfo | undefined {
+		return this.readProfileFromArgs(uri.query);
+	}
+
+	private readProfileFromArgs(query: string): IConnectionInfo | undefined {
+		if (!query) {
+			return undefined;
+		}
+
+		const args = new URLSearchParams(query);
```

Comment:
Are we planning to support connection strings?

Reply:
This is the first iteration since Benjin asked me to keep things small to avoid a high number of merge conflicts since he's still actively working on connections.

I'd say that we would want to support connection strings.